### ISSUE TITLE
Allow conditional resource creation depending on existence of external resources

### ISSUE THREAD

**Issue by maxromanovsky** [pulumi/pulumi-kubernetes#1050]
### Problem description

TL;DR

This is just a feature suggestion.

Currently, it is impossible to conditionally create resources based on other resources that are not provisioned or managed by Pulumi.

[Resource getter functions](https://www.pulumi.com/docs/intro/concepts/programming-model/#resource-get) will fail with `Preview failed: resource 'xyz' does not exist` if `<ResourceType>.get('abc', 'xyz')`  was called, but external resource does not exist.

What I propose instead is to continue program execution, but allow developer to act on result, i.e. if resource was found, then provision Pulumi-managed resources dependant on this one.

In my case, I want to create Kubernetes `ServiceMonitor` CRDs, but CRD definition is not provisioned by Pulumi. In this particular case I decided to manage that definition outside of Pulumi. And in many cases I believe similar restrictions might happen.

Question: should that issue be opened in https://github.com/pulumi/pulumi instead?

**Comment by leezen** [comment]
Thanks for filing this. This is tracked in https://github.com/pulumi/pulumi/issues/3364 -- the intention is definitely to be able to continue execution on a failed `.get` -- going to close this as a dupe unless I'm misunderstanding what you're trying to do.
